src/plugins/hexnumber/hexnumber.c

```diff
@@ -0,0 +1,275 @@
+
+	elektraFree (decValue);
+
+	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+}
+
+/**
+ * Converts a Key with a decimal number value to a Key with a hexadecimal value.
+ *
+ * @pre The key has to actually contain a decimal number string.
+ *
+ * @param key The Key whose value should be converted.
+ * @param parentKey The parent Key used to set errors.
+ *
+ * @retval #ELEKTRA_PLUGIN_STATUS_SUCCESS if the Key was successfully converted
+ * @retval #ELEKTRA_PLUGIN_STATUS_ERROR if the Key could not be converted
+ */
+static int convertDecToHex (Key * key, Key * parentKey)
+{
+	// get decimal string from key
+	const char * decValue = keyString (key);
+
+	int errnoSaved = errno;
+
+	// convert hex string to long long int
+	errno = 0;
+	char * endPtr;
+	unsigned long long int value = strtoull (decValue, &endPtr, 10);
+	if (errno == ERANGE && value == ULLONG_MAX)
+	{
+		errno = errnoSaved;
+		ELEKTRA_SET_ERRORF (ELEKTRA_ERROR_INVALID_RANGE, parentKey, "Decimal number %s out of range 0 to %llu", decValue,
+				    ULLONG_MAX);
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+	else if ((errno != 0 && value == 0) || endPtr == decValue)
+	{
+		errno = errnoSaved;
+		ELEKTRA_SET_ERRORF (ELEKTRA_ERROR_INVALID_FORMAT, parentKey, "Decimal number '%s' could not be read", decValue);
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+	errno = errnoSaved;
+
+	// convert long long int back to string (formatted as hexadecimal)
+	int result = snprintf (NULL, 0, "0x%llx", value);
+	if (result < 0)
+	{
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	size_t length = (size_t) result;
+	char * hexValue = elektraMalloc (length);
+	if (!hexValue)
+	{
+		ELEKTRA_MALLOC_ERROR (parentKey, length);
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	result = snprintf (hexValue, length, "0x%llx", value);
+	if (result < 0)
+	{
+		elektraFree (hexValue);
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	// set hex string in key
+	keySetString (key, hexValue);
+	elektraFree (hexValue);
+
+	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+}
+
+/**
+ * Checks whether a given Key's value is a hexadecimal string.
+ *
+ * @param key The Key that should be checked.
+ *
+ * @retval #true if the Key's value starts with 0x
+ * @retval #false otherwise
+ */
+static bool isHexString (const Key * key)
+{
+	return strncasecmp (keyString (key), "0x", 2) == 0;
+}
+
+/**
+ * Checks whether a given Key's type metadata is #ELEKTRA_HEXNUMBER_META_TYPE.
+ *
+ * @param key The Key that should be checked.
+ *
+ * @retval #true if the Key's type metadata is #ELEKTRA_HEXNUMBER_META_TYPE
+ * @retval #false otherwise
+ */
+static bool hasHexType (const Key * key)
+{
+	const Key * typeMeta = keyGetMeta (key, "type");
+	return typeMeta && strcmp (keyString (typeMeta), ELEKTRA_HEXNUMBER_META_TYPE) == 0;
+}
+
+/**
+ * Establish the plugin contract and convert all hexadecimal values in the KeySet to decimal.
+ *
+ * @note The plugin will attempt to convert ALL values starting with 0x from hexadecimal into decimal.
```

Comment:
Notes here will not be read by the user of the plugin.

The feature seems to dangerous, see other discussions and #1398.